### DISCUSSION TITLE
Hide edit action on table row but allow row to be clickable and open edit modal

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [filamentphp/filament#6552] [Help] [answered]
Is it possible to hide the "edit action link" on the row but still allow the row to be clickable?

![image](https://github.com/filamentphp/filament/assets/26206253/ff3a25ef-cdad-46a9-8c66-27acb7b6d421)

**Comment by lucacastelnuovo** [comment, marked as answer]
This was solved (partially) by setting `->label('')` on the edit and delete action.

```php
        //...
         ->actions([
                DeleteAction::make()
                    ->label(''),
                EditAction::make()
                    ->label(''),
            ])
```

**Reply by danharrin** [reply]
If you don't need the Delete button, you can just remove that altogether

**Reply by lucacastelnuovo** [reply]
I need the delete button. Would love to move it inside the edit form (https://github.com/filamentphp/filament/discussions/6548). I'll have to be a bit more patient for V3. So the solution is to now make the buttons less obtrusive.

If possible, I would like to hide the edit button on the row, but still, let the row be clickable to open the edit form.
If possible, I would also like to not make the row click open the edit form but still show the edit button to allow clicking on that to open the form.

**Reply by danharrin** [reply]
Instead of `label('')`, try `iconButton()`

**Reply by lucacastelnuovo** [reply]
Ah, thnx that's already a lot cleaner. Would you be able to provide a workaround for clicking on the row and not opening the editing form?
